frontend/vue/components/Statevector/StatevectorAmplitudeEncoding.vue

```diff
@@ -0,0 +1,96 @@
+<template>
+  <StatevectorBaseWidget
+    class="statevector-amplitude-encoding"
+    :number-of-elements="numberOfElements"
+    :amplitude-disk-data="amplitudeDiskData"
+  >
+    <template #default>
+      Lorem ipsum dolor sit amet, consectetur adipiscing elit. Convallis vitae elit fermentum sit. Pellentesque congue risus sed condimentum viverra dolor. Sapien nunc vulputate posuere vel.
+      <div class="statevector-amplitude-encoding__warning">
+        {{ areAllValuesZero ? 'Cannot analyze all zero values' : '&nbsp;' }}
+      </div>
+    </template>
+    <template #input-elements>
+      <SliderInput
+        v-for="(value, index) in inputData"
+        :key="index"
+        class="statevector-amplitude-encoding__input-data__slider"
+        :min="minValue"
+        :max="maxValue"
+        :step="step"
+        :value="value"
+        @value-changed="inputChangedHandler($event, index)"
+      />
+    </template>
+  </StatevectorBaseWidget>
+</template>
+
+<script lang="ts">
+import { defineComponent } from 'vue-demi'
+import { Amplitude } from '../AmplitudeDisk/amplitude'
+import SliderInput from '../common/SliderInput.vue'
+import { randomValue } from '../../../ts/utilities'
+import StatevectorBaseWidget from './StatevectorBaseWidget.vue'
+
+const numberOfElements = 4
+
+export default defineComponent({
+  name: 'StatevectorAmplitudeEncoding',
+  components: {
+    StatevectorBaseWidget,
+    SliderInput
+  },
+  props: {
+    maxValue: { type: Number, default: 1 },
+    minValue: { type: Number, default: -1 },
+    step: { type: Number, default: 0.01 }
+  },
+  data () {
+    return {
+      inputData: [] as number[],
+      areAllValuesZero: false,
+      amplitudeDiskData: [] as Amplitude[]
+    }
+  },
+  mounted () {
+    this.inputData = Array.from({ length: numberOfElements }, _ => randomValue(this.maxValue, this.minValue, this.step)) as number[]
```

Comment:
how does the `_` work here, @korgan00 ?

is this an arg?

Reply:
I use to see it to ignore the argument (in other languages). I think I can change it for `()`.